frontend/src/views/rescue/List.vue

```diff
@@ -30,18 +30,88 @@
 						type="text"
 						placeholder="Search by beneficiary name"
 						icon-left="search"
-						class="w-full rounded-full bg-white py-1 focus:bg-white"
+						class="w-full rounded-full bg-white p-1 focus:bg-white"
 						@input="(v) => (searchQuery = v)"
 						:modelValue="searchQuery"
 					/>
 
 					<Button
 						appearance="white"
 						icon="filter"
-						class="h-full rounded-full p-3"
+						class="h-full rounded-full p-4"
+						@click="showFilterDialog = !showFilterDialog"
 					></Button>
 				</div>
 
+				<div
+					v-if="showFilterDialog"
+					class="fixed inset-0 z-50 flex items-center justify-center bg-black bg-opacity-50"
+				>
+					<div
+						class="max-w-[90%] rounded-lg bg-white px-5 py-4 text-lg font-semibold text-gray-700"
+					>
+						<div class="flex justify-between">
+							<h3 class="text-xl font-semibold">Filter Rescues</h3>
+							<Button
+								@click="closeFilters"
+								icon-left="x"
+								appearance="minimal"
+								class="w-0"
+							></Button>
+						</div>
+
+						<div class="mt-4 flex flex-col space-y-2">
+							<div class="flex items-center gap-3">
+								<label for="filterGender">Gender</label>
+								<select
+									id="filterGender"
+									v-model="filterGender"
+									class="rounded-md border border-gray-300 text-lg font-semibold text-gray-700"
+								>
+									<option
+										:key="gender"
+										v-for="gender in genderOptions"
+										:value="gender"
+									>
+										{{ gender }}
+									</option>
+								</select>
+							</div>
+							<div class="flex flex-col">
+								<h3>Age</h3>
+								<div class="flex justify-center gap-0">
```

Comment:
```suggestion
								<div class="flex justify-center gap-0">
```

Does `gap-0` have any effect on UI?